write-fonts/src/tables/gvar.rs

```diff
@@ -934,4 +969,120 @@ mod tests {
+    }
+
+    #[test]
+    fn prefer_short_offsets() {
+        let _ = env_logger::builder().is_test(true).try_init();
+        const N_GLYPHS: u16 = 5;
+        let test_data = make_39_bytes_of_variation_data();
+        let a_small_number_of_variations = (0..N_GLYPHS)
+            .map(|i| GlyphVariations::new(GlyphId::new(i), test_data.clone()))
+            .collect();
+
+        let gvar = Gvar::new(a_small_number_of_variations).unwrap();
+        assert_eq!(gvar.compute_flags(), GvarFlags::empty());
+
+        let writer = gvar.compile_variation_data();
+        let mut sink = TableWriter::default();
+        writer.write_into(&mut sink);
+
+        let bytes = sink.into_data().bytes;
+        let expected_len = (N_GLYPHS + 1) as usize * u16::RAW_BYTE_LEN // offsets
+                             + 32 * N_GLYPHS as usize; // rounded size of each glyph, * 5 glyphs
+        assert_eq!(bytes.len(), expected_len);
+
+        let dumped = crate::dump_table(&gvar).unwrap();
+        let loaded = read_fonts::tables::gvar::Gvar::read(FontData::new(&dumped)).unwrap();
+
+        assert_eq!(loaded.glyph_count(), N_GLYPHS);
+        assert_eq!(loaded.flags(), GvarFlags::empty());
+        assert_eq!(
+            loaded
+                .glyph_variation_data_offsets()
+                .iter()
+                .map(|off| off.unwrap().get())
+                .collect::<Vec<_>>(),
+            vec![0u32, 32, 64, 96, 128, 160]
+        );
+    }
+
+    #[test]
+    fn use_long_offsets_when_necessary() {
+        // 2**16 * 2 / (31 + 1 padding) (bytes per tuple) = 4096 so this just overflows
+        const N_GLYPHS: u16 = 4100;
+        let _ = env_logger::builder().is_test(true).try_init();
+        let test_data = make_39_bytes_of_variation_data();
+        let a_large_number_of_variations = (0..N_GLYPHS)
+            .map(|i| GlyphVariations::new(GlyphId::new(i), test_data.clone()))
+            .collect();
+
+        let gvar = Gvar::new(a_large_number_of_variations).unwrap();
+        assert_eq!(gvar.compute_flags(), GvarFlags::LONG_OFFSETS);
+
+        let writer = gvar.compile_variation_data();
+        let mut sink = TableWriter::default();
+        writer.write_into(&mut sink);
+
+        let bytes = sink.into_data().bytes;
+        let expected_len = (N_GLYPHS + 1) as usize * u32::RAW_BYTE_LEN + 31 * N_GLYPHS as usize;
+        assert_eq!(bytes.len(), expected_len);
+
+        let dumped = crate::dump_table(&gvar).unwrap();
+        let loaded = read_fonts::tables::gvar::Gvar::read(FontData::new(&dumped)).unwrap();
+
+        assert_eq!(loaded.glyph_count(), N_GLYPHS);
+        assert_eq!(loaded.flags(), GvarFlags::LONG_OFFSETS);
+        assert_eq!(
+            loaded
+                .glyph_variation_data_offsets()
+                .iter()
+                .take(6)
+                .map(|off| off.unwrap().get())
+                .collect::<Vec<_>>(),
+            vec![0u32, 31, 62, 93, 124, 155]
+        )
```

Comment:
The majority of prefer_short_offsets and use_long_offsets_when_necessary is duplicative noise; if you diff them they differ only in a few key values. Make a helper, e.g. `assert_gvar(num glyphs, expected flags, expected offset type, expected first values)`

That will make testing the maximum number of short offsets as it's own test trivial too.